### PR TITLE
Mock borrower and better natspec

### PR DESCRIPTION
**Pull request by alcueca**

### CODE REVIEW

File: src/BaseWrapper.sol [hunk 1]
```diff
@@ -18,7 +31,9 @@ abstract contract BaseWrapper is IERC7399 {
 
     bytes internal _callbackResult;
 
-    /// @inheritdoc IERC7399
+    /// @dev The entry point for the ERC7399 flash loan. Packs data to convert the legacy flash loan into an ERC7399
+    /// flash loan. Then it calls the legacy flash loan. Once the flash loan is done, checks if there is any return
+    /// data and returns it.
```
Comment (ultrasecreth) [inline src/BaseWrapper.sol:36]: shouldn't we also keep the @inheritdoc?

File: src/BaseWrapper.sol [hunk 2]
```diff
@@ -46,10 +61,12 @@ abstract contract BaseWrapper is IERC7399 {
...
 
-    /// @dev Call the flashloan function in the child contract
+    /// @dev Call the legacy flashloan function in the child contract. This is where we borrow from Aave, Uniswap, etc.
     function _flashLoan(address asset, uint256 amount, bytes memory params) internal virtual;
 
-    /// @dev Handle the common parts of bridging the callback
+    /// @dev Handle the common parts of bridging the callback from legacy to ERC7399. Transfer the funds to the loan
+    /// receiver. Call the callback supplied by the original borrower. Approve the repayment if necessary. If there is
+    /// any result, it is kept in a storage variable to be retrieved on `flash` after the legacy flash loan is finished.
     function bridgeToCallback(address asset, uint256 amount, uint256 fee, bytes memory params) internal {
```
Comment (ultrasecreth) [inline src/BaseWrapper.sol:70]: hmm, this function name should start with a `_`

File: src/aave/AaveWrapper.sol [hunk 3]
```diff
@@ -12,6 +12,9 @@ import { FixedPointMathLib } from "lib/solmate/src/utils/FixedPointMathLib.sol";
 
 import { BaseWrapper, IERC7399, ERC20 } from "../BaseWrapper.sol";
 
+
+/// @dev Aave Flash Lender that uses Aave Pools as source of liquidity.
```
Comment (ultrasecreth) [inline src/aave/AaveWrapper.sol:16]: "Pool"

**Review by ultrasecreth — APPROVED** [review]

File: src/BaseWrapper.sol (hunk 2 above)
Comment (alcueca) [reply src/BaseWrapper.sol:70]: Fixed in #14

File: src/aave/AaveWrapper.sol (hunk 3 above)
Comment (alcueca) [reply src/aave/AaveWrapper.sol:16]: Fixed

File: src/BaseWrapper.sol (hunk 1 above)
Comment (alcueca) [reply src/BaseWrapper.sol:36]: Fixed

**Pull request merged**